lib/house.rb

```diff
@@ -0,0 +1,47 @@
+class House
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @above_market_price = "$500,000"
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def add_commas_to_price(price)
+    @price.to_s.reverse.gsub(/(\d{3})(?=\d)/, '\\1,').reverse
+  end
+
+  def above_market_price?
+    @price > @above_market_price
+  end
+
+  def rooms_from_category(category)
+    @rooms.find_all {|room| room.category == category}
+  end
+
+  def area
+    @rooms.sum {|room| room.area}
+  end
+
+  def details(price, address)
+    key_values = ["price", price ], ["address", address]
```

Comment:
This is a little bit unusual, and probably a bit harder to read than the usual way of setting key/value pairs in a hash (but kudos to you for figuring out something new!).  If you did decide to do it this way, in order to match the interaction pattern, you'd have to take the parameters out and this line would look like this instead: 

`key_values = ["price", @price ], ["address", @address]`